scanners/noopabc.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from typing import List, Type
+
+from scanners.scannerabc import ScannerABC, ScannerABCT
+
+
+###
+# Testing scan class. Does nothing time consuming or destructive,
+# but exercises many of the main hooks of domain-scan.
+class Scanner(ScannerABC):
+
+    # CSV headers for each row of data. Referenced locally.
+    headers = ["Completed", "Constant", "Variable"]
+    # Set a default number of workers for a particular scan type.
+    # Overridden by a --workers flag.
+    workers = 2  # type: int
+
+    def __init__(self, domain: str, handles: dict, environment: dict,
+                 options: dict, extra: dict={}) -> None:
+        # Optional one-time initialization per-scan. If defined, any data
+        # will be stored in the instance's ``extra_environment`` property.
+        #
+        # Run locally.
+        logging.debug("Subclass (noopabc) __init__ method for %s." % domain)
+        self.extra_environment = {"variable": domain}
+        super().__init__(domain, handles, environment, options, extra)
```

Comment:
This is more than I'm comfortable seeing scanner writers have to remember to do. The original function signature is just `init(environment, options)`. I'd prefer to see an `__init__` method can go in the base class as a non-abstract method, and `init` remain as `init(self, environment, options)`.

Reply:
I'm not sure why they have to remember any of this; the `__init__` method is in the base class and they would only need to write this for their scanner if they have per-domain functionality. Having `__init__` and `init` methods seems very confusing; if we want to take the per-domain out of `__init__`, we could move it to a method called `per_domain` or something similar.
Also, I'm assuming we would have a commented template for scanners that writers would be able to start from, rather than expecting them to remember what the methods and signatures are.